community-operators/vault/vaultoperator.v0.4.10.clusterserviceversion.yaml

```diff
@@ -0,0 +1,269 @@
+    mediatype: image/svg
+  installModes:
+  - type: OwnNamespace
+    supported: true
+  - type: SingleNamespace
+    supported: true
+  - type: MultiNamespace
+    supported: false
+  - type: AllNamespaces
+    supported: true
+  install:
+    strategy: deployment
+    spec:
+      permissions:
+      - serviceAccountName: vault-operator
+        rules:
+        - apiGroups:
+          - vault.banzaicloud.com
+          resources:
+          - vault
+          verbs:
+          - "*"
+        - apiGroups:
+          - ""
+          resources:
+          - pods
+          - services
+          - endpoints
+          - persistentvolumeclaims
+          - events
+          - configmaps
+          - secrets
+          verbs:
+          - "*"
+        - apiGroups:
+            - ""
+          resources:
+            - namespaces
+          verbs:
+            - get
+        - apiGroups:
+            - extensions
+          resources:
+            - ingresses
+          verbs:
+            - get
+            - create
+            - update
+        - apiGroups:
+            - apps
+          resources:
+            - deployments
+            - statefulsets
+      deployments:
+      - name: vault-operator
+        spec:
+          replicas: 1
+          selector:
+            matchLabels:
+              name: vault-operator-alm-owned
+          template:
+            metadata:
+              name: vault-operator-alm-owned
+              labels:
+                name: vault-operator-alm-owned
+            spec:
+              serviceAccountName: vault-operator
+              containers:
+              - name: vault-operator
+                image: banzaicloud/vault-operator:latest
+                imagePullPolicy: IfNotPresent
+                ports:
+                  - containerPort: 9091
+                    name: metrics
+                command:
+                  - vault-operator
+                livenessProbe:
+                  httpGet:
+                    path: /
+                    port: 8080
+                  initialDelaySeconds: 4
+                  periodSeconds: 10
+                  failureThreshold: 1
+                readinessProbe:
+                  httpGet:
+                    path: /ready
+                    port: 8080
+                  initialDelaySeconds: 4
+                  periodSeconds: 10
+                  failureThreshold: 1
+                env:
+                  # OPERATOR_NAMESPACE is here only for backward compatibility, the new SDK uses WATCH_NAMESPACE instead
+                  #            - name: OPERATOR_NAMESPACE
+                  #              value: ""
+                  - name: WATCH_NAMESPACE
+                    value: ""
```

Comment:
This should be set to:
```
- name: WATCH_NAMESPACE
  valueFrom:
    fieldRef:
      fieldPath: metadata.annotations['olm.targetNamespaces']
```
This is how OLM will inform the operator of which namespace(s) it should watch. In the case of the "AllNamespaces" installMode, OLM leaves this as an empty string which the operator must understand to mean as it should watch all namespaces.

Reply:
Changed that, correct, empty String means all namespaces in this operator as well.